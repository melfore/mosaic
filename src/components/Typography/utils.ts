type VariantDefaultsType = {
  [key: string]: {
    bottomSpacing: boolean;
    truncated: boolean;
  };
};

const VARIANT_DEFAULTS: VariantDefaultsType = {
  body1: {
    bottomSpacing: false,
    truncated: false,
  },
  caption: {
    bottomSpacing: false,
    truncated: false,
  },
  h5: {
    bottomSpacing: true,
    truncated: true,
  },
  h6: {
    bottomSpacing: true,
    truncated: true,
  },
  overline: {
    bottomSpacing: false,
    truncated: false,
  },
  subtitle1: {
    bottomSpacing: false,
    truncated: false,
  },
};

export const getBottomSpacing = (variant: string, bottomSpacing?: boolean): boolean =>
  bottomSpacing === undefined ? VARIANT_DEFAULTS[variant].bottomSpacing : bottomSpacing;

export const getTruncate = (variant: string, truncated?: boolean): boolean =>
  truncated === undefined ? VARIANT_DEFAULTS[variant].truncated : truncated;

export const VARIANT_COMPONENT_MAP = {
  body1: "p",
  caption: "span",
  h5: "h1",
  h6: "h2",
  overline: "h4",
  subtitle1: "h3",
};
