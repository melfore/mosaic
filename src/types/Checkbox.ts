import { BaseType } from "./Base";

export enum CheckboxSize {
  small = "small",
  default = "medium",
}

export interface CheckboxType extends BaseType {
  value?: boolean;
  onChange?: (checked: boolean) => any | void;
  required?: boolean;
  size?: CheckboxSize;
  disabled?: boolean;
  intermediate?: boolean;
}
