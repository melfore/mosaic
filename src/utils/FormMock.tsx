import React, { FC, ReactElement, useState } from "react";

interface FormMockType {
  onInputChange: Function;
  inputValue: null | number | string;
  onChangePropName?: string;
  valuePropName?: string;
}

const FormMock: FC<FormMockType> = ({
  children,
  onInputChange,
  inputValue,
  onChangePropName = "onChange",
  valuePropName = "value",
}) => {
  const [value, setValue] = useState(inputValue);
  const wrappedOnChange = (value: null | number | string) => {
    setValue(value);
    onInputChange(value);
  };

  return React.cloneElement(children as ReactElement<any>, {
    [onChangePropName]: wrappedOnChange,
    [valuePropName]: value,
  });
};

export default FormMock;
