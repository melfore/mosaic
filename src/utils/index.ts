import { ChangeEvent, MouseEvent } from "react";

export const suppressEvent = (event: ChangeEvent<any> | MouseEvent) => {
  event.preventDefault();
  event.stopPropagation();
};
