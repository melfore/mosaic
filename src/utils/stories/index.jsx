import React, { Fragment } from "react";
import CustomDocsPage from "./CustomDocsPage";

export const getDocsPageStructure = (name, blocks) => ({
  docs: {
    page: () => <CustomDocsPage blocks={blocks} name={name} />,
  },
});

export const StoriesWrapper = (props) => (
  <Fragment>
    <div className="stories-wrapper">{props.children}</div>
  </Fragment>
);
