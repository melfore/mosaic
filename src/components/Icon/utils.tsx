import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
