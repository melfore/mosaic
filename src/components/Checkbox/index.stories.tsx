import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, text, optionsKnob as options, select } from "@storybook/addon-knobs";
import { getDocsPageStructure, StoriesWrapper } from "../../utils/stories";
import Checkbox from ".";
import { CheckboxSize } from "../../types/Checkbox";
import FormMock from "../../utils/mocks/FormMock";

export default {
  title: "Checkbox",
  component: Checkbox,
  parameters: {
    ...getDocsPageStructure("Checkbox", ["usage", "canvas"]),
  },
};

export const Canvas = () => (
  // FormMock simulates external form component handling state
  // In a real case scenario "onChange" and "value" props must be passed to Checkbox
  <FormMock inputValue={boolean("value", true)} onInputChange={action("Change checkbox")}>
    <Checkbox
      dataCy={text("data-cy", "checkbox-identifier")}
      value={boolean("value", true)}
      onChange={action("Change checkbox")}
      required={boolean("value", false)}
      size={select("size", CheckboxSize, CheckboxSize.small, CheckboxSize.default)}
      disabled={boolean("disabled", false)}
      intermediate={boolean("intermediate", false)}
    />
  </FormMock>
);

export const Values = () => (
  <StoriesWrapper>
    <Checkbox dataCy={"checkbox-identifier"} value={false} onChange={() => {}} />
    <Checkbox dataCy={"checkbox-identifier"} value={true} onChange={() => {}} />
    <Checkbox dataCy={"checkbox-identifier"} value={false} intermediate onChange={() => {}} />
    <Checkbox dataCy={"checkbox-identifier"} value={true} intermediate onChange={() => {}} />
  </StoriesWrapper>
);

export const Disabled = () => (
  <StoriesWrapper>
    <Checkbox dataCy={"checkbox-identifier"} value={false} disabled />
    <Checkbox dataCy={"checkbox-identifier"} value={true} disabled />
    <Checkbox dataCy={"checkbox-identifier"} intermediate disabled />
  </StoriesWrapper>
);

export const Size = () => (
  <StoriesWrapper>
    <Checkbox dataCy={"checkbox-identifier"} value={true} />
    <Checkbox dataCy={"checkbox-identifier"} value={true} size={CheckboxSize.small} />
    <Checkbox dataCy={"checkbox-identifier"} value={false} />
    <Checkbox dataCy={"checkbox-identifier"} value={false} size={CheckboxSize.small} />
  </StoriesWrapper>
);
