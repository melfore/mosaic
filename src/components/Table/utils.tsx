import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionScope, TableActionType, TableColumnType } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: TableActionType): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    scope = TableActionScope.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    isFreeAction: scope === TableActionScope.default,
    onClick: callback,
  };
};

export const actionComponentAdapter = (props: any) => {
  const { data, action } = props;
  const { disabled, hidden, icon, isFreeAction, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {isFreeAction && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
          />
        </Fragment>
      )}
      {!isFreeAction && (
        <Fragment>
          <IconButton
            disabled={disabled}
            icon={iconName}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: TableColumnType): MTColumnType<object> => {
  const { label, path, render } = column;
  return {
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
