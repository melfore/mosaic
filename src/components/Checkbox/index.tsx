import React, { FC } from "react";
import MUICheckbox from "@material-ui/core/Checkbox";
import { BaseIntlType } from "../../types/Base";
import { CheckboxType, CheckboxSize } from "../../types/Checkbox";
import withIntl from "../../utils/hocs/withIntl";

/**
 * Checkbox component made on top of `@material-ui/core/Checkbox`.
 */
const Checkbox: FC<CheckboxType> = ({
  dataCy,
  value = false,
  onChange,
  required = false,
  size = CheckboxSize.default,
  intermediate = false,
  disabled = false,
}) => {
  const onChangeHandler = (event: any) => {
    const value = event.target.checked;
    if (onChange) {
      onChange(value);
    }
  };

  return (
    <MUICheckbox
      color="primary"
      data-cy={dataCy}
      checked={value}
      onChange={onChangeHandler}
      required={required}
      size={size}
      indeterminate={intermediate}
      disabled={disabled}
    />
  );
};

export const CheckboxIntl: FC<CheckboxType & BaseIntlType> = withIntl(Checkbox);

export default Checkbox;
