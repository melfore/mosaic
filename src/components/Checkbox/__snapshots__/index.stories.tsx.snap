// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/Checkbox Disabled smoke-test 1`] = `
<label class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
       style="margin: 0px;"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        tabindex="-1"
        aria-disabled="true"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           disabled
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
</label>
`;

exports[`Inputs/Checkbox Intermediate smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
       style="margin: 0px;"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="true"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="IndeterminateCheckBoxIcon"
    >
      <path d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z">
      </path>
    </svg>
  </span>
</label>
`;

exports[`Inputs/Checkbox Label smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mui-classname">
    Checkbox
  </span>
</label>
`;

exports[`Inputs/Checkbox LabelEnd smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mui-classname"
       data-cy="checkbox"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mui-classname">
    Checkbox
  </span>
</label>
`;

exports[`Inputs/Checkbox Localized smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="locale.checkbox"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="locale.checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="locale.checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mui-classname">
    Spunta
  </span>
</label>
`;

exports[`Inputs/Checkbox Primary smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
       style="margin: 0px;"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
</label>
`;

exports[`Inputs/Checkbox SizeSmall smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
       style="margin: 0px;"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-mui-classname"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
</label>
`;

exports[`Inputs/Checkbox Styled smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
       style="margin: 0px;"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="checkbox-check"
        style="background-color: red; color: white;"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
           checked
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxIcon"
    >
      <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
      </path>
    </svg>
  </span>
</label>
`;

exports[`Inputs/Checkbox Unchecked smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="checkbox"
       style="margin: 0px;"
>
  <span class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-mui-classname"
        data-cy="checkbox-check"
  >
    <input class="PrivateSwitchBase-input css-mui-classname"
           type="checkbox"
           data-indeterminate="false"
           data-cy="checkbox-input"
    >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="CheckBoxOutlineBlankIcon"
    >
      <path d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z">
      </path>
    </svg>
  </span>
</label>
`;
